client/jetpack-cloud/sections/agency-dashboard/sites-overview/site-add-license-notification/index.tsx

```diff
@@ -39,52 +40,80 @@ export default function SiteAddLicenseNotification() {
 		dispatch( setPurchasedLicense( license.selectedProducts.length ? license : undefined ) );
 	};
 
+	function getMessageArgs( licenses: Array< ProductInfo > ) {
+		const licenseNames = licenses.map( ( license ) => license.name );
+		const initialLicenseList = licenseNames.slice( 0, -1 );
+		const lastLicenseItem = licenseNames.slice( -1 )[ 0 ];
+		const commaCharacter = translate( ',' );
+		const conjunction =
+			licenses.length > 2
+				? translate( `%(commaCharacter)s and`, { args: { commaCharacter } } )
+				: translate( ' and' );
+		return {
+			args: {
+				selectedSite,
+				...( licenses.length > 1
+					? {
+							initialLicenseList: initialLicenseList.join( `${ commaCharacter } ` ),
+							lastLicenseItem: lastLicenseItem,
+							conjunction,
+					  }
+					: { licenseItem: lastLicenseItem } ),
+			},
+			...( licenses.length > 1 && {
+				comment: `%(initialLicenseList)s is a list of n-1 license names
```

Comment:
I am fairly sure that comments like this need to be present in the actual call to `translate()`. This is because translations are made by a static analyzer which scans the code for calls to a function called `translate()`, and not made by anything happening at runtime. I doubt that the static analyzer is complex enough to parse the code and find comments for args generated by a function like this but you can double check with the I18N team to be sure if you want.

However this function is useful for keeping the code in the `return` readable, so I suggest changing it to be something like `getTranslatedLicenseAssignmentMessage(translate, assignedLicenses)` and call that from the `return` statement. 

Then in that function every call to `translate()` should have an explicit `comment` arg set. Keep in mind that we don't know which order or even what sets of strings a particular translator will see, so for example right now they might just see a string " and" or "," with no additional context. So every string being passed to `translate` needs context such that it could be translated by itself.

Reply:
Thanks for the detailed feedback and for making me understand this. I have made the changes accordingly.